tests/main/interfaces-network-status/task.yaml

```diff
@@ -0,0 +1,69 @@
+summary: Ensure that the network-status interface works.
+
+details: |
+    The network-status interface allows a snap to monitor networking.
+
+    A snap which defines the network-status plug must be shown in the interfaces list.
+    The plug must not be autoconnected on install and, as usual, must be able to be
+    reconnected.
+
+    A snap declaring a plug on this interface must be able to monitor the network configuration
+    and ask for its status.
+
+prepare: |
+    . "$TESTSLIB/dbus.sh"
+    . "$TESTSLIB/dirs.sh"
+
+    echo "Given a snap declaring a plug on the network-status interface is installed"
+    snap install test-snapd-network-status-provider
+    snap install test-snapd-network-status-consumer
+
+    echo "And the provider dbus loop is started"
+    start_dbus_unit $SNAP_MOUNT_DIR/bin/test-snapd-network-status-provider.provider
+
+restore: |
+    rm -f net-query.output
+
+    . "$TESTSLIB/dbus.sh"
+    stop_dbus_unit
+
+execute: |
+    CONNECTED_PATTERN="^test-snapd-network-status-provider:network-status-test +test-snapd-network-status-consumer"
+    DISCONNECTED_PATTERN="^- +test-snapd-network-status-consumer:network-status-test$"
+
+    dbus_send(){
+        local method="$1"
+        echo $(dbus-send --system --print-reply --dest=com.ubuntu.connectivity1.NetworkingStatus /com/ubuntu/connectivity1/NetworkingStatus com.ubuntu.connectivity1.NetworkingStatus.${method})
+    }
+
+    echo "Connect the interface"
+    snap connect test-snapd-network-status-consumer:network-status test-snapd-network-status-provider:network-status-test
+    snap interfaces | MATCH "$CONNECTED_PATTERN"
+
+    for i in $(seq 10); do
```

Comment:
I think this loop needs fixing, if we iterate 10 times and don't match we will just carry on with the rest of the rest without an error.

Reply:
Agree, changing that.